src/attr/validators.py

```diff
@@ -614,3 +615,47 @@ def not_(validator, *, msg=None, exc_types=(ValueError, TypeError)):
     except TypeError:
         exc_types = (exc_types,)
     return _NotValidator(validator, msg, exc_types)
+
+
+@attrs(repr=False, slots=True, hash=True)
+class _OrValidator:
+    validators = attrib()
+
+    def __call__(self, inst, attr, value):
+        for v in self.validators:
+            try:
+                v(inst, attr, value)
+            except Exception:
+                continue
+            else:
+                return
```

Comment:
Hello @hynek, I wanted to ask you for your guidance here on how to approach the failing pre-commit - https://results.pre-commit.ci/run/github/29918975/1721053015.haX2lNjoRkObisfRW6gdOA - copy-pasting it here for convinience:

```
ruff.....................................................................Failed
- hook id: ruff
- exit code: 1

src/attr/validators.py:628:13: S112 `try`-`except`-`continue` detected, consider logging the exception
    |
626 |               try:
627 |                   v(inst, attr, value)
628 |               except Exception:
    |  _____________^
629 | |                 continue
    | |________________________^ S112
630 |               else:
631 |                   return
    |

src/attr/validators.py:628:13: PERF203 `try`-`except` within a loop incurs performance overhead
    |
626 |               try:
627 |                   v(inst, attr, value)
628 |               except Exception:
    |  _____________^
629 | |                 continue
    | |________________________^ PERF203
630 |               else:
631 |                   return
    |

src/attr/validators.py:628:20: BLE001 Do not catch blind exception: `Exception`
    |
626 |             try:
627 |                 v(inst, attr, value)
628 |             except Exception:
    |                    ^^^^^^^^^ BLE001
629 |                 continue
630 |             else:
    |

Found 3 errors.
```

---

Even though I generally agree with ruff's complaints, in this case, I believe all of them can be ignored.

What do you think?

---

In the case we don't want to ignore the complaints, I think the following would work:

```python

def _check_validator_is_satisfied(validator, inst, attr, value):
    try:
        v(inst, attr, value)
    except Exception:  # would still need to ignore catching naked Exception here probably?
        return False
    else:
        return True

@attrs(repr=False, slots=True, hash=True)
class _OrValidator:
    validators = attrib()

    def __call__(self, inst, attr, value):
        if any(_check_validator_is_satisfied(v, inst, attr, value) for v in self.validators):
            return

        msg = f"None of {self.validators!r} satisfied for value {value!r}"
        raise ValueError(msg)
```

Reply:
yeah just silence them (`# noqa: BLE001` etc)

Reply:
Done in 8f28bf538717c6847ba403191235351846e54b52 .